py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,524 @@
+
+            event_str['start_date'] = self._datetime_to_str(start_dt, self.format_date)
+            event_str['end_date'] = self._datetime_to_str(end_dt, self.format_date)
+
+            time_delta = self._delta_time(start_dt)
+
+            if time_delta['days'] < 0:
+                time_delta = self._delta_time(end_dt)
+                time_to_composite = self._format_timedelta(index, time_delta,
+                                                           self.format_time_left)
+            else:
+                time_to_composite = self._format_timedelta(index, time_delta,
+                                                           self.format_time_to)
+
+            self._check_warn_threshold(time_delta, event_str['start_time'],
+                                       event_str['end_time'], event_str['summary'])
+
+            curr_event_format = self._check_button_toggle(index)
+
+            if self.force_lowercase:
+                event_str = dict((k, v.lower()) if v is not None else (k, v)
+                                 for k, v in event_str.items())
+
+            event_formatted = self.py3.safe_format(
+                curr_event_format,
+                {'summary': event_str['summary'],
+                 'location': event_str['location'],
+                 'description': event_str['description'],
+                 'start_time': event_str['start_time'],
+                 'end_time': event_str['end_time'],
+                 'start_date': event_str['start_date'],
+                 'end_date': event_str['end_date'],
+                 'time_to': time_to_composite})
+
+            self.py3.composite_update(event_formatted, {'index': index})
+            responses.append(event_formatted)
+
+            index += 1
+
+        format_separator = self.py3.safe_format(self.format_separator)
+        self.py3.composite_update(format_separator, {'index': 'sep'})
+        responses = self.py3.composite_join(format_separator, responses)
+
+        return {'events': responses}
+
+    def google_calendar(self):
+        """
+        The method that outputs the response.
+
+        First, we check credential authorization. If no authorization, we display an error message,
+        and try authorizing again in 5 seconds.
+
+        Otherwise, we fetch the events, build the response, and output the resulting composite.
+        """
+        if not self.is_authorized:
+            cached_until = 0
+            self.is_authorized = self._authorize_credentials()
+
+            return {'cached_until': self.py3.time_in(cached_until),
+                    'full_text': self.py3.safe_format(self.format)}
+        else:
+            if not self.no_update:
+                self.events = self._get_events()
+
+            cached_until = self.cache_timeout
+
+            return {
+                'cached_until': self.py3.time_in(cached_until),
+                'composite': self.py3.safe_format(self.format, self._build_response())
```

Comment:
(Opinion) I don't like seeing two `return` here. See if we can use this one alone by using `composites = {}` and unident this return. There could be some other things to fix too.

Reply:
No idea why I used `command_run` there :confused: 

Yes, it's much nicer with the one return.